### PR TITLE
Replace frigate with helm-docs

### PR DESCRIPTION
**Pull request by nineinchnick**
Use `pre-commit` so developers can use a git hook to regenerate docs.

I replaced `frigate` with `helm-docs` because frigate's pre-commit hook requires `conda` to install it. `helm-docs` run from a Docker container.

Also update all GitHub actions and enable Dependabot.

### CODE REVIEW

**Review by mosabua — APPROVED** [review]
In principal I am good with this change .. however.. the rendered output does not show and description anymore.. I am hoping we can somehow make the docs for all the values more useful. Can we somehow fix this?

In fact I am contemplating adding a website similar to how we just got it added in the Trino Gateway repo .. so we can write some useful docs for the charts that go beyond what we have in the Trino docs themselves.

File: charts/trino/README.md [hunk 1]
```diff
@@ -1,119 +1,120 @@
...
+| coordinator.resources | object | `{}` |  |
+| coordinator.secretMounts | list | `[]` |  |
+| coordinator.terminationGracePeriodSeconds | int | `30` |  |
+| coordinator.tolerations | list | `[]` |  |
+| env | list | `[]` |  |
+| envFrom | list | `[]` |  |
+| eventListenerProperties | object | `{}` |  |
+| image.digest | string | `""` |  |
+| image.pullPolicy | string | `"IfNotPresent"` |  |
+| image.registry | string | `""` |  |
```
Comment (mosabua) [inline charts/trino/README.md:52]: This is an example where description is missing.

**Comment by mosabua** [comment]
Also .. if helm-docs is good .. we want to look at adding its usage for the Trino Gateway chart in the trino-gateway repo.

File: charts/trino/README.md (hunk 1 above)
Comment (nineinchnick) [reply charts/trino/README.md:52]: Looks like `helm-docs` process descriptions that start with a double dash. It's documented here: https://github.com/norwoodj/helm-docs?tab=readme-ov-file#valuesyaml-metadata

Weirdly, I don't see a way to disable/configure this. I don't really want to change all descriptions unless there's no other option. In that case, maybe it'll be a good chance to review them.

Comment (mosabua) [reply charts/trino/README.md:52]: There are only a few there at the moment .. I have a task on my backlog to clean this up. All of them should have some description in my opinion .. and then the docs site we add should build it up more to create reasonable examples for different use cases...

Comment (nineinchnick) [reply charts/trino/README.md:52]: Yeah, that's hardcoded in the tool: https://github.com/norwoodj/helm-docs/blob/4dc6510f8eafae294891bead0a4f4df5e49ad1c4/pkg/helm/chart_info.go#L18

I'll review all existing descriptions and let's see how it'll look.

Comment (nineinchnick) [reply charts/trino/README.md:52]: Another question - there's a flag to select sort order, do we want alphabetical or the order from `values.yaml`?

**Comment by nineinchnick** [comment]
@mosabua I've used a custom template to print all values as a definition list instead of a table. I included those large examples that wouldn't fit in the table well. I also filled in most descriptions with links to Kubernetes and Trino documentation, where relevant. Please review.

File: charts/trino/README.md.gotmpl [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{ template "chart.header" . }}
```
Comment (nineinchnick) [inline charts/trino/README.md.gotmpl:1]: I think I should add a `.helmignore` file so this template is not packaged in the chart, or put it in a different location and point to it when calling `helm-doc` from `.pre-commit-config.yaml`.

**Review by mosabua — APPROVED** [review]
This is a massive improvement. I think we should merge and then we can refine more going forward in terms of specific wording other nits in separate PRs.

File: charts/trino/README.md.gotmpl (hunk 2 above)
Comment (mosabua) [reply charts/trino/README.md.gotmpl:1]: That sounds good too .. you can do that in this or a separate PR.

**Pull request merged**